state/application.go

```diff
@@ -3544,3 +3544,23 @@ func appUnitNames(st *State, appName string) ([]string, error) {
 	}
 	return unitNames, nil
 }
+
+// UploadedCharmOrigin looks up the application that references the provided
+// charm URL and returns the origin from where it was originally downloaded.
```

Comment:
Correct. This is called from the download code when the blob is already stored to return back the origin that corresponds to the stored blob (rather than what the caller requested as an origin). I will drop the _originally_ bit to avoid any confusion.